### PR TITLE
Adds some Fish Content

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds a new biotype, MOB_AQUATIC, indicating the mob is water-themed somehow. Given to carp, lobstrosities, frogs, axolotls, penguins, fire sharks.

Aquatic mobs can be hooked by fishing rods, even without a jawed fishing hook installed.

The carp and fish infusion sets now give the infusee the aquatic biotype. Added support for infusions adding a biotype.

You can check for a fish's pulse with a stethoscope, which will tell you its status even without fishing skill.

Refined fish health status checks to be more precise.

Added 'Fishy' Reagent, a version of strange reagent that only works for fish or aquatic biotype mobs. It's made with omnizine, saltwater, and carpotoxin or tetrodotoxin.

Added a lifish chemical reaction that creates fish.
## Why It's Good For The Game

Fish content fish content fish content

> Adds a new biotype, MOB_AQUATIC, indicating the mob is water-themed somehow. Given to carp, lobstrosities, frogs, axolotls, penguins, fire sharks.

We were really missing this one by now.

> Aquatic mobs can be hooked by fishing rods, even without a jawed fishing hook installed.

> The carp and fish infusion sets now give the infusee the aquatic biotype. Added support for infusions adding a biotype.

I want to reel in fish people. This is going to be hilarious.

> You can check for a fish's pulse with a stethoscope, which will tell you its status even without fishing skill.

Fish doctor content - also lets you see the exact status of a fish's health even if you haven't interacted w/ fishing that shift.

> Added 'Fishy' Reagent, a version of strange reagent that only works for fish or aquatic biotype mobs. It's made with omnizine, saltwater, and carpotoxin or tetrodotoxin.

A somewhat easier to get version of strange reagent, purely for fish, as they die very easily and the road to making SR is a large diversion. I might add this to cargo, maybe?

> Added a lifish chemical reaction that creates fish.

Fish. fish fish fish fish fish.

@Ghommie 
## Changelog
:cl:
add: Adds a new biotype, MOB_AQUATIC, indicating the mob is water-themed somehow. Given to carp, lobstrosities, frogs, axolotls, penguins, fire sharks.
add: Aquatic mobs can be hooked by fishing rods, even without a jawed fishing hook installed.
add: The carp and fish infusion sets now give the infusee the aquatic biotype. Added support for infusions adding a biotype.
add: You can check for a fish's pulse with a stethoscope, which will tell you its status even without fishing skill.
qol: Refined fish health status checks to be more precise.
add: Added 'Fishy' Reagent, a version of strange reagent that only works for fish or aquatic biotype mobs. It's made with omnizine, saltwater, and carpotoxin or tetrodotoxin.
add: Added a lifish chemical reaction that creates fish.
/:cl:

### CODE REVIEW

**Comment by DrAmazing343** [comment]
Fish Doctor content slowly creeping into the game will never not be funny, and ditto for more fish content in general. Excellent PR!

File: code/modules/fishing/fish/_fish.dm [hunk 1]
```diff
@@ -434,27 +452,47 @@
...
 	if(HAS_MIND_TRAIT(user, TRAIT_EXAMINE_FISH))
 		. += span_notice("It's [size] cm long.")
 		. += span_notice("It weighs [weight] g.")
-		if(HAS_TRAIT(src, TRAIT_FISHING_BAIT))
-			. += span_smallnoticeital("It can be used as a fishing bait.")
+
+	. += get_health_warnings(user, always_deep = FALSE)
+
+	if(HAS_TRAIT(src, TRAIT_FISHING_BAIT))
+		. += span_smallnoticeital("It can be used as a fishing bait.")
```
Comment (00-Steven) [inline code/modules/fishing/fish/_fish.dm:462]: seems to be moved from behind a `HAS_MIND_TRAIT(user, TRAIT_EXAMINE_FISH)` check up a few lines to here, not behind one
as this isn't in the pr body, is this intentional?

File: code/modules/fishing/fish/_fish.dm [hunk 2]
```diff
@@ -434,27 +452,47 @@
...
+
 	if(bites_amount)
 		. += span_warning("It's been bitten by someone.")
 
+/obj/item/fish/proc/get_health_warnings(mob/user, always_deep = FALSE)
+	if(!HAS_MIND_TRAIT(user, TRAIT_EXAMINE_DEEPER_FISH) && !always_deep)
+		return
+	if(status == FISH_DEAD)
+		. += span_deadsay("It's [HAS_MIND_TRAIT(user, TRAIT_NAIVE) ? "taking the big snooze" : "dead"].")
+		return .
```
Comment (00-Steven) [inline code/modules/fishing/fish/_fish.dm:472]: we're not adding to a list here so no need for this
```suggestion
		return span_deadsay("It's [HAS_MIND_TRAIT(user, TRAIT_NAIVE) ? "taking the big snooze" : "dead"].")
```

File: code/modules/fishing/fish/_fish.dm [hunk 3]
```diff
@@ -434,27 +452,47 @@
...
+			warnings += "very unhealthy"
+		if(0.5 to 0.75)
+			warnings += "unhealthy"
+		if(0.75 to 0.9)
+			warnings += "mostly healthy"
+
+	if(length(warnings))
+		. += span_warning("It's [english_list(warnings)].")
+
+	return .
```
Comment (00-Steven) [inline code/modules/fishing/fish/_fish.dm:494]: same here- we're not adding anything before this either so, may as well
```suggestion
	if(!length(warnings))
		return

	return span_warning("It's [english_list(warnings)].")
```

**Review by 00-Steven — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fish/_fish.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/fishing/fish/_fish.dm:462]: honestly, sure

File: code/datums/elements/organ_set_bonus.dm [hunk 4]
```diff
@@ -80,10 +84,21 @@
...
+	// Add traits
 	if(length(bonus_traits))
 		owner.add_traits(bonus_traits, REF(src))
+
+	// Add biotype
+	if(owner.mob_biotypes & bonus_biotype)
+		biotype_added = FALSE
+		return TRUE
+	owner.mob_biotypes |= bonus_biotype
+	biotype_added = TRUE
```
Comment (MrMelbert) [inline code/datums/elements/organ_set_bonus.dm:96]: So if the mob already has the biotype we don't add anything else? No limb overlays, no activation text? Why?

File: code/datums/elements/organ_set_bonus.dm [hunk 5]
```diff
@@ -96,10 +111,18 @@
 /datum/status_effect/organ_set_bonus/proc/disable_bonus()
 	SHOULD_CALL_PARENT(TRUE)
 	bonus_active = FALSE
+
+	// Remove traits
 	if(length(bonus_traits))
 		owner.remove_traits(bonus_traits, REF(src))
+	// Remove biotype (if added)
+	if(biotype_added == TRUE)
```
Comment (MrMelbert) [inline code/datums/elements/organ_set_bonus.dm:119]:
```suggestion
	if(biotype_added)
```

File: code/datums/elements/organ_set_bonus.dm [hunk 6]
```diff
@@ -57,6 +57,10 @@
 	var/required_biotype = MOB_ORGANIC
 	/// A list of traits added to the mob upon bonus activation, can be of any length.
 	var/list/bonus_traits = list()
+	/// Bonus biotype to add on bonus activation.
+	var/bonus_biotype
+	/// If the biotype was added - used to check if we should remove the biotype or not, on organ set loss.
+	var/biotype_added = FALSE
```
Comment (MrMelbert) [inline code/datums/elements/organ_set_bonus.dm:63]: Roach infusion gives `MOB_BUG`, refactor it to use this sytem

File: code/datums/elements/organ_set_bonus.dm (hunk 4 above)
Comment (carlarctg) [reply code/datums/elements/organ_set_bonus.dm:96]: whoops

File: code/modules/fishing/fish/_fish.dm [hunk 7]
```diff
@@ -207,6 +207,9 @@
 	if(HAS_TRAIT(source, TRAIT_CATCH_AND_RELEASE))
 		context[SCREENTIP_CONTEXT_RMB] = "Release"
 		return CONTEXTUAL_SCREENTIP_SET
+	if(istype(held_item, /obj/item/clothing/neck/stethoscope))
+		context[SCREENTIP_CONTEXT_LMB] = "Check Pulse"
+		return CONTEXTUAL_SCREENTIP_SET
```
Comment (Ghommie) [inline code/modules/fishing/fish/_fish.dm:212]: This is in `add_item_context`, which is for when src is the held item and hovering another object. Move this to `add_context`

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 8]
```diff
@@ -208,6 +208,16 @@
...
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_HEALING | REACTION_TAG_PLANT | REACTION_TAG_OTHER
 
+/datum/chemical_reaction/medicine/fishy_reagent
+	results = list(/datum/reagent/medicine/strange_reagent/fishy_reagent = 3)
+	required_reagents = list(/datum/reagent/medicine/omnizine = 1, /datum/reagent/water/salt = 1, /datum/reagent/toxin/carpotoxin = 1)
+	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_HEALING | REACTION_TAG_OTHER
+
+/datum/chemical_reaction/medicine/fishy_reagent/alt
+	results = list(/datum/reagent/medicine/strange_reagent/fishy_reagent = 3)
+	required_reagents = list(/datum/reagent/medicine/omnizine = 1, /datum/reagent/water/salt = 1, /datum/reagent/toxin/tetrodotoxin = 1)
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/recipes/medicine.dm:218]: I suppose the required TTX could be half as much given it's a bit rarer than carpotoxin.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 9]
```diff
@@ -575,6 +575,19 @@
...
+/datum/chemical_reaction/lifish
+	required_reagents = list(/datum/reagent/medicine/strange_reagent/fishy_reagent = 1, /datum/reagent/medicine/c2/synthflesh = 1, /datum/reagent/blood = 1)
+	required_temp = 374
+	reaction_flags = REACTION_INSTANT
+	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_UNIQUE
+
+/datum/chemical_reaction/lifish/on_reaction(datum/reagents/holder, datum/equilibrium/reaction, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i in 1 to rand(1, created_volume)) // More flop.
+		var/obj/item/fish/spawned_fish = pick(subtypesof(/obj/item/fish))
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/recipes/others.dm:587]: I feel like this shouldn't spawn holographic fish or McGill, but you do what you do.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 10]
```diff
@@ -575,6 +575,19 @@
...
+		new spawned_fish(location)
+	..()
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/recipes/others.dm:589]:
```suggestion
	return ..()
```

File: code/__DEFINES/mobs.dm [hunk 11]
```diff
@@ -73,6 +73,8 @@
 #define MOB_PLANT (1 << 10)
 ///The mob is a goopy creature, probably coming from xenobiology.
 #define MOB_SLIME (1 << 11)
+/// Mob is fish or fish-related.
```
Comment (Ghommie) [inline code/__DEFINES/mobs.dm:76]:
```suggestion
/// Mob is fish or water-related.
```
It's a tad broader than just fish if frogs and penguin also have it.

File: code/modules/reagents/chemistry/recipes/medicine.dm (hunk 8 above)
Comment (carlarctg) [reply code/modules/reagents/chemistry/recipes/medicine.dm:218]: made it fgive more

File: code/modules/fishing/fish/_fish.dm (hunk 3 above)
Comment (00-Steven) [reply code/modules/fishing/fish/_fish.dm:493]: did you forget to address this or?

File: code/modules/fishing/fish/types/mining.dm [hunk 12]
```diff
@@ -143,6 +143,9 @@
 /obj/item/fish/boned/make_edible(weight_val)
 	return //it's all bones and no meat.
 
+/obj/item/fish/bones/get_health_warnings(mob/user, always_deep = FALSE)
```
Comment (carlarctg) [inline code/modules/fishing/fish/types/mining.dm:146]:
```suggestion
/obj/item/fish/boned/get_health_warnings(mob/user, always_deep = FALSE)
```

**Pull request merged**